File: test/net/dinkla/raytracer/objects/CompoundTest.java
package net.dinkla.raytracer.objects;

import net.dinkla.raytracer.math.BBox;
import net.dinkla.raytracer.math.MathUtils;
import net.dinkla.raytracer.objects.compound.Compound;
import org.junit.Test;

import static org.junit.Assert.assertEquals;

/**
 * Created by IntelliJ IDEA.
 * User: Jörn Dinkla
 * Date: 14.05.2010
 * Time: 21:17:44
 * To change this template use File | Settings | File Templates.
 */
public class CompoundTest {

    @Test
    public void testGetBoundingBox() throws Exception {
        Sphere s = new Sphere(1.0);
        Compound c = new Compound();
        c.add(s);

        BBox bboxC = c.getBoundingBox();
        BBox bboxS = s.getBoundingBox();

        assertEquals(bboxC.getP(), bboxS.getP().minus(MathUtils.INSTANCE.getK_EPSILON()));
        assertEquals(bboxC.getQ(), bboxS.getQ().plus(MathUtils.INSTANCE.getK_EPSILON()));
    }

}

File: src/test/kotlin/net/dinkla/raytracer/objects/CompoundTest.kt
package net.dinkla.raytracer.objects

import net.dinkla.raytracer.math.MathUtils
import net.dinkla.raytracer.objects.compound.Compound
import org.junit.jupiter.api.Test

import org.junit.jupiter.api.Assertions.assertEquals


class CompoundTest {

    @Test
    @Throws(Exception::class)
    fun testGetBoundingBox() {
        val s = Sphere(radius = 1.0)
        val c = Compound()
        c.add(s)

        val bboxC = c.boundingBox
        val bboxS = s.boundingBox

        assertEquals(bboxC.p, bboxS.p.minus(MathUtils.K_EPSILON))
        assertEquals(bboxC.q, bboxS.q.plus(MathUtils.K_EPSILON))
    }

}
